### PR TITLE
KOGITO-3743: Online Editor - Open from source can't open GitHub URLs

### PR DESCRIPTION
**Pull request by ljmotta**
## Task
https://issues.redhat.com/browse/KOGITO-3743

This PR fixes some functionallities that were introduced on https://github.com/kiegroup/kogito-tooling/pull/289 

## Test
Some URLs can be opened on the `Open from Source` functionality and all of them should be working properly. Here are some examples:
- GitHub Raw: https://raw.githubusercontent.com/ljmotta/bpmn-dmn-examples/master/sample.bpmn
- GitHub Normal: https://github.com/ljmotta/bpmn-dmn-examples/blob/master/sample.bpmn
- Gist Raw: https://gist.githubusercontent.com/ljmotta/45e76c172c596397e02c6cef4aae28b5/raw/3be5cde9f1605c04e017503ca77ba0f08a922117/Drawing%25201-3.dmn
- Gist Normal: https://gist.github.com/ljmotta/45e76c172c596397e02c6cef4aae28b5
- Gist Specific: https://gist.github.com/ljmotta/db1c79d9919ffa3eb40ad8b7cada5b7f#file-test1000-bpmn

### CODE REVIEW

File: packages/online-editor/src/common/GithubTokenModal.tsx [hunk 1]
```diff
@@ -52,15 +58,26 @@ export function GithubTokenModal(props: Props) {
...
+    setErrorStatus(false);
   }, []);
 
+  const validated = useMemo(() => {
+    if (errorStatus) {
+      return "error";
+    }
+    return "default";
+  }, [errorStatus]);
+
```
Comment (tiagobento) [inline packages/online-editor/src/common/GithubTokenModal.tsx]: Good ternary candidate.

File: packages/online-editor/src/common/GithubService.ts [hunk 2]
```diff
@@ -90,7 +90,12 @@ export class GithubService {
     const testOctokit = new Octokit({ auth: token });
     return testOctokit.users
       .getAuthenticated()
-      .then(res => Promise.resolve(res.data.login))
+      .then(res => {
+        if ((res.headers as any)["x-oauth-scopes"].split(", ").indexOf("gist") > -1) {
+          return Promise.resolve(res.data.login);
+        }
+        return Promise.resolve(undefined);
+      })
```
Comment (tiagobento) [inline packages/online-editor/src/common/GithubService.ts]: Do we need Promise.resolve() wrapping our return values here?

**Review by tiagobento — COMMENTED** [review]
@ljmotta Thanks for the PR! Minor observations inline.

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me. Thanks :slightly_smiling_face:

**Review by tiagobento — APPROVED** [review]

**Pull request merged**